### PR TITLE
Docs refresh

### PR DESCRIPTION
**Pull request by lovelydinosaur**
* Push back on "performance". A bit silly really. Was important at one point in order to demonstrate that the ASGI approach was not any *worse* than tightly integrated alternatives. But we've established that point now.
* Drop "Support for HTTP/2 is planned".
* Add "Why ASGI" section to better provide the rationale for the ASGI ecosystem.
* Include "Alternative ASGI servers" on the README (as well as the existing section on the homepage.)
* Drop "Designed & built in Brighton, England." - I think there's some validity to design approaches that have a strong initial author voice, because it helps establish a clear direction and design intent, but it's also diminishing to the *vast* number of folks who've worked on it over the years, and the multitude of contributions of all different kinds that make it possible.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -79,8 +79,63 @@ $ uvicorn example:app
...
-<p align="center"><i>Uvicorn is <a href="https://github.com/encode/uvicorn/blob/master/LICENSE.md">BSD licensed</a> code.<br/>Designed & built in Brighton, England.</i><br/>&mdash; 🦄  &mdash;</p>
+## Why ASGI?
+
+Most well established Python Web frameworks started out as WSGI-based frameworks.
+
+WSGI applications are a single, synchronous callable that takes a request and returns a response.
+This doesn’t allow for long-lived connections, like you get with long-poll HTTP or WebSocket connections,
+which WSGI doesn't support well.
+
+Having a async concurrency model also allows for options such as lightweight background tasks,
```
Comment (aminalaee) [inline README.md]: `an async` ?

**Review by aminalaee — COMMENTED** [review]
There's an extra file added `docs/.DS_Store`.

File: README.md [hunk 2]
```diff
@@ -79,8 +79,63 @@ $ uvicorn example:app
...
+and can be less of a limiting factor for endpoints that have long periods being blocked on network
+I/O such as dealing with slow HTTP requests.
+
+---
+
+## Alternative ASGI servers
+
+A strength of the ASGI protocol is that it decouples the server implementation
+from the application framework. This allows for an ecosystem of interoperating
+webservers and applcation frameworks.
```
Comment (florimondmanca) [inline README.md]: (Typo)

```suggestion
webservers and application frameworks.
```

**Review by florimondmanca — COMMENTED** [review]
Makes complete sense. :+1:

File: docs/index.md [hunk 3]
```diff
@@ -422,8 +418,26 @@ async def app(scope, receive, send):
...
+and can be less of a limiting factor for endpoints that have long periods being blocked on network
+I/O such as dealing with slow HTTP requests.
+
+---
+
 ## Alternative ASGI servers
 
+A strength of the ASGI protocol is that it decouples the server implementation
+from the application framework. This allows for an ecosystem of interoperating
+webservers and applcation frameworks.
```
Comment (lovelydinosaur) [inline docs/index.md]:
```suggestion
webservers and application frameworks.
```

File: README.md (hunk 1 above)
Comment (lovelydinosaur) [inline README.md]:
```suggestion
Having an async concurrency model also allows for options such as lightweight background tasks,
```

File: docs/index.md [hunk 4]
```diff
@@ -422,8 +418,26 @@ async def app(scope, receive, send):
...
 
+## Why ASGI?
+
+Most well established Python Web frameworks started out as WSGI-based frameworks.
+
+WSGI applications are a single, synchronous callable that takes a request and returns a response.
+This doesn’t allow for long-lived connections, like you get with long-poll HTTP or WebSocket connections,
+which WSGI doesn't support well.
+
+Having a async concurrency model also allows for options such as lightweight background tasks,
```
Comment (lovelydinosaur) [inline docs/index.md]:
```suggestion
Having an async concurrency model also allows for options such as lightweight background tasks,
```

**Comment by lovelydinosaur** [comment]
Thanks folks! Comments addressed.
I've also bumped up https://github.com/encode/starlette/pull/1477 which is similar.

File: README.md [hunk 5]
```diff
@@ -79,8 +79,63 @@ $ uvicorn example:app
...
+$ hypercorn app:App
+```
+
+### Mangum
+
+[Mangum][mangum] is an adapter for using ASGI applications with AWS Lambda & API Gateway.
+
+---
+
+<p align="center"><i>Uvicorn is <a href="https://github.com/encode/uvicorn/blob/master/LICENSE.md">BSD licensed</a> code.<br/>Designed and crafted with care.</i><br/>&mdash; 🦄  &mdash;</p>
```
Comment (lovelydinosaur) [inline README.md]:
```suggestion
<p align="center"><i>Uvicorn is <a href="https://github.com/encode/uvicorn/blob/master/LICENSE.md">BSD licensed</a> code.<br/>Designed & crafted with care.</i><br/>&mdash; 🦄  &mdash;</p>
```

**Review by florimondmanca — APPROVED** [review]

**Pull request merged**